kraken/node_actions/aws_node_scenarios.py

```diff
@@ -0,0 +1,177 @@
+        self.boto_client.reboot_instances(
+            InstanceIds=[instance_id]
+        )
+
+
+    def wait_until_running(self, instance_id):
+        self.boto_instance.wait_until_running(
+            InstanceIds=[instance_id]
+        )
+
+
+    def wait_until_stopped(self, instance_id):
+        self.boto_instance.wait_until_stopped(
+            InstanceIds=[instance_id]
+        )
+
+
+    def wait_until_terminated(self, instance_id):
+        self.boto_instance.wait_until_terminated(
+            InstanceIds=[instance_id]
+        )
+
+
+class aws_node_scenarios(abstract_node_scenarios):
+    def __init__(self, node_scenario):
+        self.instance_kill_count = node_scenario["instance_kill_count"]
+        self.node_name = node_scenario["node_name"]
+        self.label_selector = node_scenario["label_selector"]
+        self.timeout = node_scenario["timeout"]
+        self.aws = AWS()
+
+
+    def node_start_scenario(self):
+        for _ in range(self.instance_kill_count):
+            try:
+                logging.error("Starting node_start_scenario injection")
+                node = get_node(self.node_name, self.label_selector)
+                instance_id = self.aws.get_instance_id(node)
+                logging.info("Starting the node %s with instance ID: %s " % (node, instance_id))
+                self.aws.start_instances(instance_id)
+                self.aws.wait_until_running(instance_id)
+                runcommand.invoke("kubectl wait --for=condition=Ready node/" + node + " --timeout=" + str(self.timeout) + "s")
+                logging.info("Node with instance ID: %s is in running state" % (instance_id))
+                logging.info("node_start_scenario has been successfully injected!")
+            except Exception as e:
+                logging.error("Failed to start node instance. Encountered following exception: %s. Test Failed" % (e))
+                logging.error("node_start_scenario injection failed!")
+                sys.exit(1)
+
+
+    def node_stop_scenario(self):
+        for _ in range(self.instance_kill_count):
+            try:
+                logging.error("Starting node_stop_scenario injection")
+                node = get_node(self.node_name, self.label_selector)
+                instance_id = self.aws.get_instance_id(node)
+                logging.info("Stopping the node %s with instance ID: %s " % (node, instance_id))
+                self.aws.stop_instances(instance_id)
+                self.aws.wait_until_stopped(instance_id)
+                logging.info("Node with instance ID: %s is in stopped state" % (instance_id))
+                wait_for_unknown_status(node, self.timeout)
+            except Exception as e:
+                logging.error("Failed to stop node instance. Encountered following exception: %s. Test Failed" % (e))
+                logging.error("node_stop_scenario injection failed!")
+                sys.exit(1)
+
+
+    def node_start_stop_scenario(self):
```

Comment:
The node_start_scenario and stop functions can be reused by passing the node_name/selector to act on instead of repeating the logic. Thoughts?